add SB-POSIX to ASD dependencies

paulapatience opened the pull request:
Also guarded the SBCL dependencies behind the SBCL feature, as is done further below in the ASD file (though there remain various unguarded calls to SBCL packages throughout the source code), and removed the comment beside the SB-MPFR dependency, because it is used in more than just tests (in MPD-PI via MPD-RECIPROCAL).

stylewarning commented:
I believe these dependencies can be guarded with specific ASDF syntax. Something [like this](https://github.com/fare/asdf/blob/master/doc/best_practices.md#conditional-code).

stylewarning commented:
(Side note: honestly, I should probably cordon off the TEST-PI into the test package where MPFR is expected. The main package isn't the right place I don't think. Just laziness on my part.)

paulapatience committed:
add SB-POSIX to ASD dependencies

Also guarded the SBCL dependencies behind the SBCL feature, as is done
further below in the ASD file (though there remain various unguarded
calls to SBCL packages throughout the source code), and removed the
comment beside the SB-MPFR dependency, because it is used in more than
just tests (in MPD-PI via MPD-RECIPROCAL).

paulapatience commented:
> I believe these dependencies can be guarded with specific ASDF syntax. Something [like this](https://github.com/fare/asdf/blob/master/doc/best_practices.md#conditional-code).

I knew about `:if-feature`, but not about `:feature` for `:depends-on`. Thanks.

I've force-pushed the PR with `:feature :sbcl` instead of `#+sbcl`.

stylewarning commented:
Thanks!
